mailchimp_ecommerce.module

```diff
@@ -533,6 +535,7 @@ function mailchimp_ecommerce_delete_cart($cart_id) {
       /* @var \Mailchimp\MailchimpEcommerce $mc_ecommerce */
       $mc_ecommerce = mailchimp_get_api_object('MailchimpEcommerce');
       $mc_ecommerce->deleteCart($store_id, $cart_id);
+      _mailchimp_ecommerce_remove_order($cart_id, $store_id);
```

Comment:
looks good. my only question is if we have the remove order handling correct. Do we want to remove the local order on cart delete?

Reply:
I would, because the cart will need to be added again to MailChimp. So, if the record still exists locally but the Cart no longer exists in MCAPI, we'll run into the add vs. update problem again. Since the modules would believe it already exists, they'd want to call `updateCart()` when they would really need to call `addCart().`

Perhaps this is just a naming issue; I kept everything as `_order()` because carts and orders are interchangeable in these modules, really, even though MailChimp thinks of them separately (and MC Orders have more properties than Carts do). 

But I also don't want to necessarily discern between the two in our database if our only real purpose is to decide which API methods we want to call to make sure our data is accurate there.